### PR TITLE
Add score matrix size errors

### PR DESCRIPTION
**Pull request by atadkase**

### CODE REVIEW

File: cudapoa/src/application_parameters.cpp [hunk 1]
```diff
@@ -109,6 +111,17 @@ ApplicationParameters::ApplicationParameters(int argc, char* argv[])
...
 
+    if (!banded && adaptive)
+    {
+        throw std::runtime_error("adaptive alignment cannot run with full alignment");
+    }
+
+    if (adaptive && default_bandwidth)
+    {
+        // Bump up the default bandwidth as it is 256 for static banded which is too low
+        band_width = 2048;
```
Comment (atadkase) [inline cudapoa/src/application_parameters.cpp:122]: I added this bump according to our previous discussion but it is not needed for the sample-xyz.txt files.

Comment (r-mafi) [reply cudapoa/src/application_parameters.cpp:122]: it helps to clarify the role of `band_width` in adaptive-banded alignment. A note to indicate this parameter is used to determine maximum score matrix size. I think 2048 might still be conservative, but we can have a better understanding of this once we check the results from benchmark tests.

File: cudapoa/src/cudapoa_kernels.cuh [hunk 2]
```diff
@@ -284,6 +286,7 @@ __global__ void generatePOAKernel(uint8_t* consensus_d,
                                                                                             sequence,
                                                                                             seq_len,
                                                                                             scores,
+                                                                                            banded_window_scores_size,
```
Comment (r-mafi) [inline cudapoa/src/cudapoa_kernels.cuh:289]: I'd change it to `banded_score_matrix_size`

File: cudapoa/src/cudapoa_nw_adaptive_banded.cuh [hunk 3]
```diff
@@ -363,6 +365,12 @@ __device__ void set_band_parameters(ScoreT* scores,
 
     // update head_index for the nex row
     head_index += static_cast<int64_t>(band_width + CUDAPOA_BANDED_MATRIX_RIGHT_PADDING);
+    if (head_index > scores_size)
```
Comment (r-mafi) [inline cudapoa/src/cudapoa_nw_adaptive_banded.cuh:368]: should be changed to `>=`, but it's super low risk!

**Review by r-mafi — APPROVED** [review]

**Comment by r-mafi** [comment]
I just made few minor comments, I will address those comments after merging.

**Pull request merged**